go/worker/txnscheduler/committee/node.go

```diff
@@ -128,6 +136,59 @@ func (n *Node) HandlePeerMessage(ctx context.Context, message *p2p.Message) (boo
 	return false, nil
 }
 
+// CheckTx checks the given call in the node's runtime.
+func (n *Node) CheckTx(ctx context.Context, call []byte) error {
+	if n.commonNode.CurrentBlock == nil {
```

Comment:
Accessing the `CurrentBlock` requires holding the CrossNode lock if called outside of methods which assume the caller does so (e.g., the ones ending in `Locked`).